src/commands/client-kill.json

```diff
@@ -41,77 +41,89 @@
         ],
         "arguments": [
             {
-                "name": "ip:port",
-                "type": "string",
-                "optional": true
-            },
-            {
-                "token": "ID",
-                "name": "client-id",
-                "type": "integer",
-                "optional": true,
-                "since": "2.8.12"
-            },
-            {
-                "token": "TYPE",
-                "name": "normal_master_slave_pubsub",
+                "name": "filter",
                 "type": "oneof",
-                "optional": true,
-                "since": "2.8.12",
                 "arguments": [
                     {
-                        "name": "normal",
-                        "type": "pure-token",
-                        "token": "normal"
-                    },
-                    {
-                        "name": "master",
-                        "type": "pure-token",
-                        "token": "master",
-                        "since": "3.2.0"
-                    },
-                    {
-                        "name": "slave",
-                        "type": "pure-token",
-                        "token": "slave"
+                        "name": "old-format",
```

Comment:
looking at this again, maybe it should say `ip:port` instead of old-format?
seems more useful and less confusing than just saying old format.

also, maybe we should mark this pattern ad deprecated.
IIRC a mechanism we added in #10545 but didn't use yet.

Reply:
Alright, I changed the name back to ip:port. 

I also added a deprecation notice. I put in 7.0.4 for now. Not sure which version you want to put there. Does your policy allow deprecation on patch releases? Or does it need to be at least a minor release?

Reply:
this is just metadata (which i bet no one uses yet).
we did add marker to deprecate HMSET since version 4.0.
so can you please check when the alternative syntax was added and use that?

Reply:
Do you mean the new format with alternative syntax? If yes, according to https://redis.io/commands/client-kill/, that would be 2.8.12.

Reply:
yes, `CLIENT KILL ADDR`.
pity that unlike commands, for args we don't have the `replaced_by` feature.

Reply:
I can open a new issue proposing the addition of replaced_by for arguments and link this here to add this information when That gets added.

I also want to put the description of the name field in the ring here https://redis.io/docs/reference/command-arguments/.
I would argue that having three arguments with the name ip:port in a single command spoof the "identification purpose" that is described there. In reality it seems you need name and token to identify a argument. But that is probably something for a separate issue as well.

Reply:
@guybe7 @itamarhaber can you pleases respond to the above. 
i do remember we had some concerns and a plan, but i don't remember the details.

Reply:
@valkum 
1. please open an issue for having a "replaced_by" for arguments
2. regarding the argument identification, IIRC we decided that we would use the indices of the arg. for example, the old-format ip:addr is arguments[0].arguments[0] and LADDR is arguments[0].arguments[1].arguments[4]
perhaps it's not so straightforward, we could also add a mandatory "id" field to each arg, but that's a lot of work (in this case maybe rename "name" to "display"?)

@itamarhaber thoughts?

Reply:
The doc is outdated and therefore wrong - https://github.com/redis/redis-doc/pull/2048 fixes the fallacy.

Agreeing w/ no. 1.
Adding an 'id' field (and perhaps renaming 'name' to 'display') sounds like a significant chore especially if we want to have meaningful ids, but if it makes life easier down the road, I'm for it.

Reply:
so we don't currently have a proper way to refer to a specific argument (e.g. `id`)?
i.e. if we have a documentation block that wants to explain an arg in detail and its relation to other args, it needs to refer to them by their name or ID...
i thought we did have something for that, i vaguely remember some PR by Itamar dealing with this dilemma.